source/multigrid/mg_transfer_matrix_free.cc

```diff
@@ -112,7 +117,20 @@ void MGTransferMatrixFree<dim,VectorType>::build
                                                    dirichlet_indices,
                                                    weights_unvectorized,
                                                    this->copy_indices_global_mine,
-                                                   this->ghosted_level_vector);
```

Comment:
It is true that the `setup_transfer` does two operations with `ghosted_level_vector`, the initialization and the computation of weights. However, there is a subtle issue that happens in that function, too: The code generates indices in local MPI space, and creates a vector partitioner that matches this local numbering. I do not really understand what kind of modification you made here, but that might be somewhere to look.